client/pages/components/EmailHeader.js

```diff
@@ -1,43 +1,88 @@
-/** EmailHeaeder.js
+/** EmailHeader.js
  * Displays name and subject of email.
  * OnClick will redirect to ViewEmail with emailID
  * @author Nicholas Morash (A00378981)
+ * @author Jay Patel (A00433907)
+ * @author Justin Gray
  */
-
 import React from "react";
 import { useRouter } from "next/router";
 import * as Formatter from "../../utils/formatter";
+import axios from "axios";
+import defaults from "../../utils/defaults";
 
+// userAuthor: true if user is the author of the email (sender)
 export default function EmailHeader({ userAuthor, email }) {
-    const router = useRouter();
+  const router = useRouter();
 
-    let author = "";
-    if (userAuthor && email) author = Formatter.contacts(email.to)
-    if (email) author = Formatter.contact(email.from);
+  let author = "";
+  if (userAuthor && email) author = Formatter.contacts(email.to);
+  if (email) author = Formatter.contact(email.from);
 
-    function formatSubject() {
-        const sub = email ? email.subject : "";
-        if(sub.length >= 20){
-            return sub.substring(0, 20) + "...";
-        }
-        return sub;
+  function formatSubject() {
+    const sub = email ? email.subject : "";
+    if (sub.length >= 20) {
+      return sub.substring(0, 20) + "...";
     }
+    return sub;
+  }
 
-    function handleClick() {
-        router.push(`/ViewEmail`); //todo dynamic route
-        // localStorage.setItem('emailID', email._id);
-        localStorage.setItem('emailData', JSON.stringify(email));
-        // store.dispatch({
-        //     type: "userAuthor",
-        //     payload: {userAuthor}
-        // })
-    }
+  function handleClick() {
+    router.push(`/ViewEmail`); //todo dynamic route
+    localStorage.setItem("emailData", JSON.stringify(email));
+  }
 
-    return (
-        <div onClick={handleClick}>
-            <span>{author}</span>
-            <span>{formatSubject()}</span>
-            <span></span>
+  // delete an email from the proper box
+  function handleDelete() {
+    const jwt = localStorage.getItem("token");
+    axios
+      .delete(
+        `${defaults.serverUrl}/email/${userAuthor ? "sent" : "inbox"}/${
+          email._id
+        }`,
+        {
+          headers: {
+            "x-auth-token": jwt,
+          },
+        }
+      )
+      .then((res) => {
+        // locally delete the email, since it no longer exists on the server
+        window.location.reload();
```

Comment:
I remember I added this as a bandaid solution. Ideally we should modify the state of the application, and this should trigger a re-render of the appropriate components. Refreshing a react application is not great. Especially since it completely wipes the state.

Is there a way we can improve the state and remove this refresh?